tools/testbench/topology.c

```diff
@@ -734,6 +734,13 @@ int parse_topology(struct sof *sof, struct shared_lib_table *library_table,
 			temp_comp_list = (struct comp_info *)
 					 realloc(temp_comp_list, size);
 
+			if (!temp_comp_list && size) {
+				free(hdr);
+				fclose(file);
```

Comment:
if temp_comp_list was previously allocated will the memory still be allocated even though the pointer was lost when realloc fails?

Reply:
Yes. From manual page of realloc: **If realloc() fails, the original block is left untouched; it is not freed or moved.**

I will use a temporary pointer for that location and then when there's an error, i will free the already allocated memory using that temporary pointer.

Reply:
Updated with the suggestions.